roles/elasticsearch/tasks/main.yml

```diff
@@ -1,16 +1,199 @@
 ---
+- name: check if certs exist
+  stat: 'path=/etc/letsencrypt/live/{{ hostname }}/privkey.pem'
+  register: certfile
+
+- name: fail if certs not present
+  fail: 'msg="/etc/letsencrypt/live/{{ hostname }}/privkey.pem not found"'
+  when: not certfile.stat.exists
+
+- name: set vars
+  ansible.builtin.set_fact:
+    elastic_running: false
+    cluster_running: false
+
+- name: check if elasticsearch already running
+  ansible.builtin.uri:
+    url: 'https://{{ hostname }}:{{ http_port }}/'
+    user: '{{ admin_user }}'
+    password: '{{ admin_password }}'
+    validate_certs: false
+    return_content: true
+  register: elastic_response
+  ignore_errors: true
+
+- name: set a var if ES is already running
+  when: "elastic_response.content is defined and 'cluster_name' in elastic_response.content"
+  ansible.builtin.set_fact:
+    elastic_running: true
+
 - name: install elasticsearch rpm gpg key
+  when: elastic_running is false
   rpm_key: state=present key=https://artifacts.elastic.co/GPG-KEY-elasticsearch
 
 - name: install elasticsearch 8.x repository
   copy: src=elasticsearch-8.repo dest=/etc/yum.repos.d/elasticsearch.repo owner=root group=root mode=0644
-  when: elasticsearch_version == 8
+  when: elasticsearch_version == 8 and elastic_running is false
 
 - name: install elasticsearch and dependencies
+  when: elastic_running is false
   yum: name={{item}} state=latest
   with_items:
     - elasticsearch
   notify: update aide
 
+- name: open http port
+  ansible.builtin.shell: 'firewall-cmd --zone=public --add-port={{ http_port }}/tcp'
+
+- name: open transport port
+  ansible.builtin.shell: 'firewall-cmd --zone=public --add-port={{ transport_port }}/tcp'
+
```

Comment:
This needs to use the firewalld module and be applied immediately and permanently

Reply:
Done